syft/pkg/cataloger/dotnet/config.go

```diff
@@ -0,0 +1,255 @@
+	Providers            []string                  `yaml:"package-providers,omitempty" json:"package-providers,omitempty" mapstructure:"package-providers"`
+	ProviderCredentials  []nugetProviderCredential `yaml:"package-provider-credentials,omitempty" json:"package-provider-credentials,omitempty" mapstructure:"package-provider-credentials"`
+}
+
+// DefaultCatalogerConfig create a CatalogerConfig with default options, which includes:
+// - setting the default remote package providers if none are provided
+func DefaultCatalogerConfig() CatalogerConfig {
+	g := CatalogerConfig{
+		SearchLocalLicenses:  false,
+		SearchRemoteLicenses: false,
+		Providers:            []string{},
+		ProviderCredentials:  []nugetProviderCredential{},
+	}
+
+	nuget := os.Getenv("NUGET_SEARCH_LOCAL_LICENSES")
+	if value, err := strconv.ParseBool(nuget); err == nil {
+		g = g.WithSearchLocalLicenses(value)
+	}
+
+	remote := os.Getenv("NUGET_SEARCH_REMOTE_LICENSES")
+	if value, err := strconv.ParseBool(remote); err == nil {
+		g = g.WithSearchRemoteLicenses(value)
+	}
+
+	// process the Nuget package repository settings (for remote search)
+	nugetProviders := os.Getenv("NUGET_PACKAGE_PROVIDERS")
+	if nugetProviders == "" {
+		nugetProviders = getDefaultProviders()
+	}
+	g = g.WithProviders(nugetProviders)
+
+	// process the Nuget package repository credential settings (for remote search)
+	g = g.WithCredentials(os.Getenv("NUGET_PACKAGE_PROVIDER_CREDENTIALS"))
+
+	return g
+}
+
+func (g CatalogerConfig) WithSearchLocalLicenses(input bool) CatalogerConfig {
+	g.SearchLocalLicenses = input
+	return g
+}
+
+func (g CatalogerConfig) WithSearchRemoteLicenses(input bool) CatalogerConfig {
+	g.SearchRemoteLicenses = input
+	return g
+}
+
+func (g CatalogerConfig) WithProviders(input string) CatalogerConfig {
+	if input == "" {
+		return g
+	}
+	g.Providers = strings.Split(input, ",")
+	return g
+}
+
+func (g CatalogerConfig) WithCredentials(input string) CatalogerConfig {
+	if input == "" {
+		return g
+	}
+	g.ProviderCredentials = []nugetProviderCredential{}
+	credentials := strings.Split(input, ",")
+	for _, credential := range credentials {
+		if credentialParts := strings.Split(credential, ":"); len(credentialParts) == 2 {
+			g.ProviderCredentials = append(g.ProviderCredentials, nugetProviderCredential{
+				Username: credentialParts[0],
+				Password: credentialParts[1],
+			})
+		}
+	}
+	return g
+}
+
+type sourceAPIResource struct {
+	ID      string `json:"@id"`
+	Type    string `json:"@type"`
+	Comment string `json:"comment,omitempty"`
+}
+
+type sourceAPIContext struct {
+	Vocab   string `json:"@vocab"`
+	Comment string `json:"comment,omitempty"`
+}
+
+type sourceAPI struct {
+	Version   string              `json:"version"`
+	Resources []sourceAPIResource `json:"resources"`
+	Context   *sourceAPIContext   `json:"@context,omitempty"`
+}
+
+func determineDotnetExecutablePath() (string, error) {
```

Comment:
Today, Syft does not shell out to any tools (with one very specific exception of `cosign`). And while we have talked about using tools if they are found, we haven't pulled the trigger on this anywhere yet and _very much want to avoid executing tools_, if we can.

Are there some specific files that this configuration is present we could use instead of shelling out to `dotnet`?

Reply:
I don't like the necessity either, but so far I have unfortunately not found a way around it. ☹️